source/elements/oneTBB/source/named_requirements.rst

```diff
@@ -35,20 +46,50 @@ For example, the table below shows pseudo-signatures for a *sortable* type ``T``
 
 .. cpp:function:: bool operator<(const T& x, const T& y)
 
-  Compare x and y.
+  Compare ``x`` and ``y``.
 
 .. cpp:function:: void swap(T& x, T& y)
 
-  Swap x and y.
+  Swap ``x`` and ``y``.
 
 ---------------------------------------------------------------------------------------------
 
-A real signature may differ from the pseudo-signature that it implements in ways where implicit
-conversions would deal with the difference. For an example type ``U``, the real signature that
-implements ``operator<`` in the table above can be expressed as ``int operator<( U x, U y )``,
-because C++ permits implicit conversion from ``int`` to ``bool``, and implicit conversion from ``U``
-to (``const U&``). Similarly, the real signature ``bool operator<( U& x, U& y )`` is acceptable
-because C++ permits implicit addition of a const qualifier to a reference type.
+For a given type ``U``, the real signature that implements ``operator<`` in the table above 
+can be expressed as ``int operator<( U x, U y )``, because C++ permits implicit conversion from
+``int`` to ``bool``, and implicit conversion from ``const U&`` to ``U`` if the type is copy-constructible.
+As a counter-example, the real signature ``bool operator<( U& x, U& y )`` is not acceptable
+because C++ does not permit implicit removal of the ``const`` qualifier from a type, and so the code
+would not compile if the implementation attempts to pass a constant object to the function.
+
+Besides pseudo-signatures, semantic requirements also need to be met by real types and functions.
+For example, while ``std::pair<U, U> swap(U x, U y)`` fits the pseudo-signature for *Sortable*
+via implicit conversion of references to values and implicit drop of the returned value
+(ignored by a library implementation), it is unable to swap the actual variables passed to the function
+and therefore does not meet the semantic requirements of *Sortable*.
+
+The following table provides guidance for the types of parameters used in pseudo-signatures.
+
+==========================  ================================  =============================
+Pseudo-signature parameter  General semantics                 Alternative real parameters
+==========================  ================================  =============================
+``const T& x``              The function is not supposed      - ``T x``
+                            to modify the argument.           - ``U& x``
```

Comment:
I am a bit confused. In one of the preceded paragraphs, there is a counter-example where type `U&` is not suitable, but here it is mentioned as a possible type of a real parameter.

Reply:
Below is a) an explanation that `U` is a template parameter type, and b) a note that the applicable alternatives may depend on semantic requirements. In other words, the table does not mean that an alternative is always correct to use.